### PR TITLE
Migrate the student history exam tables to the new table implementation.

### PR DESCRIPTION
**Pull request by wtritch**
This PR migrates the student history ica/iab/sum tables to a single table using the new turbo table implementation.

I also took the opportunity to replace some usages of the deprecated AssessmentType enum with assessment type codes.

image

image

image

### CODE REVIEW

File: webapp/src/main/webapp/src/app/student/results/student-results.component.html [hunk 1]
```diff
@@ -51,8 +51,9 @@ <h3>{{'student-results.title' | translate}}</h3>
   <ng-container *ngFor="let assessmentType of assessmentTypes">
     <ng-container *ngFor="let subject of getSubjectsForType(assessmentType)">
       <div class="well wide-content-container">
-        <span class="h3 label-group {{ colorService.getColor(assessmentType) }}">
-          <span class="label">{{assessmentType | assessmentType}}</span><span class="label">{{subject}}</span>
+        <span class="h3 label-group"
+              [ngClass]="getAssessmentTypeColor(assessmentType)">
+          <span class="label">{{'common.assessment-type.' + assessmentType + '.short-name' | translate}}</span><span class="label">{{subject}}</span>
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/results/student-results.component.html]: If there's any break between these two <span> tags it creates a space in the [IAB|Math] pill box =/

File: webapp/src/main/webapp/src/app/student/results/student-results.component.html [hunk 2]
```diff
@@ -63,18 +64,13 @@ <h3>{{'student-results.title' | translate}}</h3>
...
-            [student]="examHistory.student"
-            [minimumItemDataYear]="minimumItemDataYear"
-            [exams]="examsByTypeAndSubject.get(assessmentType).get(subject)"></student-history-ica-summative-table>
+        <div class="mt-md"
+             [hidden]="isCollapsed(assessmentType, subject)">
+          <student-history-table [assessmentType]="assessmentType"
+                                 [student]="examHistory.student"
+                                 [minimumItemDataYear]="minimumItemDataYear"
+                                 [exams]="examsByTypeAndSubject.get(assessmentType).get(subject)">
+          </student-history-table>
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/results/student-results.component.html]: The one-true-student-history-table.

File: webapp/src/main/webapp/src/app/student/results/student-results.component.ts [hunk 3]
```diff
@@ -15,6 +14,12 @@ import { AssessmentSubjectType } from "../../shared/enum/assessment-subject-type
 import { Utils } from "../../shared/support/support";
 import { ReportingEmbargoService } from "../../shared/embargo/reporting-embargo.service";
 import { ApplicationSettingsService } from '../../app-settings.service';
+import { Comparator, ranking } from "@kourge/ordering/comparator";
+import { AssessmentDefinition } from "../../aggregate-report/assessment/assessment-definition";
+
+const AssessmentTypes: string[] = [ "iab", "ica", "sum" ];
+const AssessmentTypeColorOrder: string[] = [ "ica", "iab", "sum"];
+const TypeDisplayOrder: Comparator<string> = ranking(AssessmentTypes);
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/results/student-results.component.ts]: I dislike that our color ordering is different from our display ordering... I also dislike that we have the set of all assessment types defined in here (twice.)  But I'm not sure what a clean alternative is.  Thoughts?

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html [hunk 4]
```diff
@@ -1,71 +0,0 @@
-<p-dataTable [value]="exams"
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: no more iab-only table

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.html [hunk 5]
```diff
@@ -1,101 +0,0 @@
-<!-- Table Display State Toggle -->
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.html]: no more ica/summative-only table

File: webapp/src/main/webapp/src/primeng-overrides.less [hunk 6]
```diff
@@ -19,6 +19,10 @@ body {
 // TODO Added .ui-table* to apply styles to primeng 5 turbo table.
 // TODO Once we move all tables to p-table we can remove the .ui-datatable* selectors here
 // Tables
+.ui-table-auto-layout > .ui-table-wrapper {
+  overflow: visible;
+}
```
Comment (wtritch) [inline webapp/src/main/webapp/src/primeng-overrides.less]: With our oversized tables, the new table style was hiding the last item's popup menu inside a funky internal scroll pane.

**Comment by wtritch** [comment]
Merging this now, will resolve any comments as they arrive.

File: webapp/src/main/webapp/src/app/student/results/student-results.component.html (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/student/results/student-results.component.html]: that's html for ya : j

File: webapp/src/main/webapp/src/app/student/results/student-results.component.ts [hunk 7]
```diff
@@ -100,20 +98,20 @@ export class StudentResultsComponent implements OnInit {
       || this.filterState.subject != params.subject;
   }
 
-  isCollapsed(assessmentType: AssessmentType, subject: string): boolean {
+  isCollapsed(assessmentType: string, subject: string): boolean {
     return this.displayState[ assessmentType ][ subject ].collapsed;
   }
 
-  toggleCollapsed(assessmentType: AssessmentType, subject: string): void {
+  toggleCollapsed(assessmentType: string, subject: string): void {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/student/results/student-results.component.ts]: 👍

File: webapp/src/main/webapp/src/primeng-overrides.less (hunk 6 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/primeng-overrides.less]: 8 .

**Pull request merged**